app/src/main/java/com/project/app/base/BaseFragment.kt

```diff
@@ -0,0 +1,158 @@
+package com.project.app.base
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import android.view.*
+import android.view.inputmethod.InputMethodManager
+import android.widget.ImageButton
+import android.widget.TextView
+import android.widget.Toast
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.viewbinding.ViewBinding
+import com.project.app.R
+import com.project.app.helpers.LoadingDialog
+import com.project.app.helpers.ToolbarType
+import kotlinx.coroutines.Job
+
+abstract class BaseFragment<VB : ViewBinding> : Fragment(), IBaseFragment {
+    lateinit var activityIndicator: LoadingDialog
+    private var job = Job()
+    private lateinit var topBarLayout: ConstraintLayout
+    lateinit var leftButtonToolbar: ImageButton
+    private lateinit var textViewMenuTitle: TextView
+    private lateinit var rigthButtonToolbar: ImageButton
+
+    private lateinit var fragmentView: View
+
+    private var _binding: ViewBinding? = null
+    abstract val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> VB
+    @Suppress("UNCHECKED_CAST")
+    val binding: VB
+        get() = _binding as VB
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = bindingInflater.invoke(inflater, container, false)
+        return requireNotNull(_binding).root
```

Comment:
Acá, si queremos evitar el requireNotNull, creo que podemos usar el null safety de Kotlin. Que sería más o menos así...

```suggestion
        return _binding!!.root
```

Y también arrojaría un null pointer exception si esto es nulo. Aunque todo también depende de si cambiamos el uso del binding como mencionaba en anteriores comentarios. Lo podemos ver en conjunto.

Reply:
Fue otro agregado de los foros que leí con respecto a la migración. En el ondestroy vuelve a ser nulo siempre, si queres probamos el cambio, no debería afectar igual porque en el siempre se instancia

Reply:
Podemos explorarlo. Pero tengo entendido que es básicamente lo mismo. Solo que el requireNotNull es de java y el `!!` es de Kotlin. Pero hacen lo mismo. Solo sería algo un poquito más simplificado.